backend/optimize.ml

```diff
@@ -0,0 +1,210 @@
+[@@@ocaml.warning "+a-29-40-41-42-4"]
+
+module DLL = Flambda_backend_utils.Doubly_linked_list
+
+module type IntCell = sig
+  include Csv_data.Cell
+
+  val update : t -> t
+end
+
+module IntCell : IntCell = struct
+  type t = int
+
+  let to_string = Int.to_string
+
+  let empty () = 0
+
+  let update t = t + 1
+end
+
+module IntCsv = Csv_data.Make (IntCell)
+
+let csv_singleton = ref Option.None
+
+let random_hex_string () =
+  let random_value = Random.int (1 lsl 24) in
+  Printf.sprintf "%06x" random_value
```

Comment:
I think having something deterministic,
yet with statistically no collisions, would
be useful. In the past, I have used the
digest of the concatenation of `Sys.argv`.